source/codegen/metadata/nidaqmx/functions.py

```diff
@@ -6957,7 +6992,6 @@
         'returns': 'int32'
     },
     'GetExtendedErrorInfo': {
-        'codegen_method': 'private',
```

Comment:
Removing codegen_method=private from GetExtendedErrorInfo adds this method to nidaqmx.proto. We shouldn't do that because this function must be called on the same thread that produced the error, and gRPC clients can't control which server thread handles each RPC request.

@IPetersNI @reckenro Was this file edited in https://github.com/ni/grpc-device/pull/755 without updating scrapigen?

We probably need a special case in scrapigen to fix this.

Reply:
Can we override this in https://github.com/ni/grpc-device/blob/main/source/codegen/metadata/nidaqmx/functions_addon.py ?

Reply:
I think it was. David suggested a follow up with [this comment](https://github.com/ni/grpc-device/pull/755#issuecomment-1267934620) to update scrapigen but I'm guessing that didn't happen (I didn't follow it after that comment):
> > ... thoughts on if we should update scrapigen for this (I'm guessing yes) and possibly more info on the how.

> Yes, we should. If it would work to just remove the function, then the best path forward is to copy the function from DAQmx.h into DAQmx_exlcude.h. But if it's important to keep it present but mark it private, we'd write special code to do that in src/scrapigen/daqapi/filterfunctions.py.

As for if we want to just patch this grpc-device side (to not leak this into nidaqmx-python) I'm not sure if adding it to functions_addon.py would work. At least currently, those addon files are only used for metdata_validation suppressions. @astarche , does that sound right? I don't remember us actually plugging those `_override_metadata` dictionaries in anywhere.

Reply:
That's correct. Personally I would add a new tag and have a `metadata_mutation` in `grpc-device` that inserts `codegen_method: private` when that flag is set. Naming is always a challenge for these things. `relies_on_thread_local_state: True` or `is_get_last_error_method: True`. Maybe something like `remote: False` or "remotable: False`, but those are pretty generic and I'm not sure if remotable is a word ...

Reply:
Added a fix in https://github.com/ni/grpc-device-scrapigen/pull/127/commits/2ff22d5b025bf6287e43aff545216a1ba34eca4a to mark the methods private.

Reference: https://github.com/ni/grpc-device-scrapigen/blob/main/src/scrapigen/transform/transformfunctions.py#L458